### PR TITLE
[TT-4756] Refactor oidc auth OAS conversion

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3950/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2022670304)

File: apidef/oas/security.go [hunk 1]
```diff
@@ -267,6 +269,80 @@ func (s *OAS) extractOAuthTo(api *apidef.APIDefinition, name string) {
...
+	authConfig := apidef.AuthConfig{DisableHeader: true}
+
+	oidc := s.getTykAuthentication().OIDC
+	api.UseOpenID = oidc.Enabled
+	oidc.AuthSources.ExtractTo(&authConfig)
+
+	api.OpenIDOptions.SegregateByClient = oidc.SegregateByClientId
+
+	for _, p := range oidc.Providers {
+		clientIDs := make(map[string]string)
```
Comment (titpetric) [inline apidef/oas/security.go:331]: hint length with len(p.ClientToPolicyMapping) (prealloc)

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3950/merge
Commit: bda02d8d8c3f6caf0ce02fd4c9385faac5eefd1f 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2028893952)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3950/merge
Commit: 87f31ed6d0b6ac8676416912c650271e6aae39de 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2029027861)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 518b3f8a932ecee5030d89ca693e89e6a5d316a7 Refactor oidc auth OAS conversion (#3950)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2029213504)

**Comment by furkansenharputlu** [comment]
Manually merged to `release-4`

**Pull request merged**